From d4c57e5b4a70ede44e8e1f0f5039decc6e43070d Mon Sep 17 00:00:00 2001
From: Alex Gonzalez <alex.gonzalez@digi.com>
Date: Wed, 20 Mar 2019 17:58:23 +0100
Subject: [PATCH] ccimx6ul: environment: Add max_beb_per1024 parameter to
 kernel cmd line

This allows to attach small partitions. However, it would be preferable
to redesign with fewer partitions and more volumes.

Signed-off-by: Alex Gonzalez <alex.gonzalez@digi.com>
---
 include/configs/ccimx6ul_common.h |  5 +++++
 include/configs/ccimx6ulsbc.h     | 14 +++++++++++---
 include/configs/ccimx6ulstarter.h | 14 +++++++++++---
 scripts/config_whitelist.txt      |  5 +++++
 4 files changed, 32 insertions(+), 6 deletions(-)

diff --git a/include/configs/ccimx6ul_common.h b/include/configs/ccimx6ul_common.h
index 540c9a7a1f8b..27f103091613 100644
--- a/include/configs/ccimx6ul_common.h
+++ b/include/configs/ccimx6ul_common.h
@@ -216,6 +216,11 @@
 #define CONFIG_ENV_MTD_RECOVERY_INDEX	"4"
 #define CONFIG_ENV_MTD_ROOTFS_INDEX	"5"
 #define CONFIG_ENV_MTD_UPDATE_INDEX	"6"
+#define CONFIG_ENV_MTD_SAFE_MAX_BEB_PER1024		"5"
+#define CONFIG_ENV_MTD_LINUX_MAX_BEB_PER1024		"5"
+#define CONFIG_ENV_MTD_RECOVERY_MAX_BEB_PER1024	"5"
+#define CONFIG_ENV_MTD_ROOTFS_MAX_BEB_PER1024		"5"
+#define CONFIG_ENV_MTD_UPDATE_MAX_BEB_PER1024		"5"
 #define CONFIG_ENV_MTD_SETTINGS		"mtdids=" MTDIDS_DEFAULT "\0"
 
 /* Max percentage of reserved blocks for bad block management per partition */
diff --git a/include/configs/ccimx6ulsbc.h b/include/configs/ccimx6ulsbc.h
index 8a5b45437297..dae78550d3b9 100644
--- a/include/configs/ccimx6ulsbc.h
+++ b/include/configs/ccimx6ulsbc.h
@@ -185,9 +185,12 @@
 	CONFIG_ENV_MTD_SETTINGS \
 	"bootargs_linux=\0" \
 	"bootargs_nand_linux=setenv bootargs console=${console},${baudrate} " \
-		"${bootargs_linux} ${mtdparts} ubi.mtd=${mtdsafeindex} " \
-		"ubi.mtd=${mtdlinuxindex} ubi.mtd=${mtdrecoveryindex} " \
-		"ubi.mtd=${mtdrootfsindex} ubi.mtd=${mtdupdateindex} " \
+		"${bootargs_linux} ${mtdparts} " \
+		"ubi.mtd=${mtdsafeindex},0,${mtdsafemaxbebper1024} " \
+		"ubi.mtd=${mtdlinuxindex},0,${mtdlinuxmaxbebper1024} " \
+		"ubi.mtd=${mtdrecoveryindex},0,${mtdrecoverymaxbebper1024} " \
+		"ubi.mtd=${mtdrootfsindex},0,${mtdrootfsmaxbebper1024} " \
+		"ubi.mtd=${mtdupdateindex},0,${mtdupdatemaxbebper1024} " \
 		"root=ubi3_0 " \
 		"rootfstype=ubifs rw " \
 		"${bootargs_once} ${extra_bootargs}\0" \
@@ -210,6 +213,11 @@
 	"mtdrecoveryindex=" CONFIG_ENV_MTD_RECOVERY_INDEX "\0" \
 	"mtdrootfsindex=" CONFIG_ENV_MTD_ROOTFS_INDEX "\0" \
 	"mtdupdateindex=" CONFIG_ENV_MTD_UPDATE_INDEX "\0" \
+	"mtdsafemaxbebper1024=" CONFIG_ENV_MTD_SAFE_MAX_BEB_PER1024 "\0" \
+	"mtdlinuxmaxbebper1024=" CONFIG_ENV_MTD_LINUX_MAX_BEB_PER1024 "\0" \
+	"mtdrecoverymaxbebper1024=" CONFIG_ENV_MTD_RECOVERY_MAX_BEB_PER1024 "\0" \
+	"mtdrootfsmaxbebper1024=" CONFIG_ENV_MTD_ROOTFS_MAX_BEB_PER1024 "\0" \
+	"mtdupdatemaxbebper1024=" CONFIG_ENV_MTD_UPDATE_MAX_BEB_PER1024 "\0" \
 	"recoverycmd=" \
 		"setenv mtdbootpart " CONFIG_RECOVERY_PARTITION ";" \
 		"boot\0" \
diff --git a/include/configs/ccimx6ulstarter.h b/include/configs/ccimx6ulstarter.h
index 0a1f85d8d86b..dba6e3ed59ab 100644
--- a/include/configs/ccimx6ulstarter.h
+++ b/include/configs/ccimx6ulstarter.h
@@ -185,9 +185,12 @@
 	CONFIG_ENV_MTD_SETTINGS \
 	"bootargs_linux=\0" \
 	"bootargs_nand_linux=setenv bootargs console=${console},${baudrate} " \
-		"${bootargs_linux} ${mtdparts} ubi.mtd=${mtdsafeindex} " \
-		"ubi.mtd=${mtdlinuxindex} ubi.mtd=${mtdrecoveryindex} " \
-		"ubi.mtd=${mtdrootfsindex} ubi.mtd=${mtdupdateindex} " \
+		"${bootargs_linux} ${mtdparts} " \
+		"ubi.mtd=${mtdsafeindex},0,${mtdsafemaxbebper1024} " \
+		"ubi.mtd=${mtdlinuxindex},0,${mtdlinuxmaxbebper1024} " \
+		"ubi.mtd=${mtdrecoveryindex},0,${mtdrecoverymaxbebper1024} " \
+		"ubi.mtd=${mtdrootfsindex},0,${mtdrootfsmaxbebper1024} " \
+		"ubi.mtd=${mtdupdateindex},0,${mtdupdatemaxbebper1024} " \
 		"root=ubi3_0 " \
 		"rootfstype=ubifs rw " \
 		"${bootargs_once} ${extra_bootargs}\0" \
@@ -210,6 +213,11 @@
 	"mtdrecoveryindex=" CONFIG_ENV_MTD_RECOVERY_INDEX "\0" \
 	"mtdrootfsindex=" CONFIG_ENV_MTD_ROOTFS_INDEX "\0" \
 	"mtdupdateindex=" CONFIG_ENV_MTD_UPDATE_INDEX "\0" \
+	"mtdsafemaxbebper1024=" CONFIG_ENV_MTD_SAFE_MAX_BEB_PER1024 "\0" \
+	"mtdlinuxmaxbebper1024=" CONFIG_ENV_MTD_LINUX_MAX_BEB_PER1024 "\0" \
+	"mtdrecoverymaxbebper1024=" CONFIG_ENV_MTD_RECOVERY_MAX_BEB_PER1024 "\0" \
+	"mtdrootfsmaxbebper1024=" CONFIG_ENV_MTD_ROOTFS_MAX_BEB_PER1024 "\0" \
+	"mtdupdatemaxbebper1024=" CONFIG_ENV_MTD_UPDATE_MAX_BEB_PER1024 "\0" \
 	"recoverycmd=" \
 		"setenv mtdbootpart " CONFIG_RECOVERY_PARTITION ";" \
 		"boot\0" \
diff --git a/scripts/config_whitelist.txt b/scripts/config_whitelist.txt
index f5f2505119f7..546cb22f26cf 100644
--- a/scripts/config_whitelist.txt
+++ b/scripts/config_whitelist.txt
@@ -995,11 +995,16 @@ CONFIG_ENV_IS_NOWHERE
 CONFIG_ENV_MAX_ENTRIES
 CONFIG_ENV_MIN_ENTRIES
 CONFIG_ENV_MTD_LINUX_INDEX
+CONFIG_ENV_MTD_LINUX_MAX_BEB_PER1024
 CONFIG_ENV_MTD_RECOVERY_INDEX
+CONFIG_ENV_MTD_RECOVERY_MAX_BEB_PER1024
 CONFIG_ENV_MTD_ROOTFS_INDEX
+CONFIG_ENV_MTD_ROOTFS_MAX_BEB_PER1024
 CONFIG_ENV_MTD_SAFE_INDEX
+CONFIG_ENV_MTD_SAFE_MAX_BEB_PER1024
 CONFIG_ENV_MTD_SETTINGS
 CONFIG_ENV_MTD_UPDATE_INDEX
+CONFIG_ENV_MTD_UPDATE_MAX_BEB_PER1024
 CONFIG_ENV_OFFSET
 CONFIG_ENV_OFFSET_OOB
 CONFIG_ENV_OFFSET_REDUND
